### ISSUE TITLE
Koltin and List<String> type converter

### ISSUE THREAD

**Issue by akapuscinski** [Raizlabs/DBFlow#1354]
DBFlow Version: 4.0.4
Issue Kind (Bug, Question, Feature): Bug, Question

I have a data class defined like this:
```
@Table(database = StopsDb::class)
data class FavouriteStop(
        @PrimaryKey @Column var id: String = "",
        @Index @Column var name: String = "",
        @Column(typeConverter = StringListConverter::class) var directions: List<String> = listOf(),
        @Column(typeConverter = StringListConverter::class) var selectedDirections: List<String> = listOf()
) : BaseRXModel()
```
and as I don't want to create a separate table only to store Strings I created a List<String> Type converter like this:

```
class StringListConverter : TypeConverter<String, List<String>>() {

    val separator = ","

    override fun getDBValue(model: List<String>?): String {
        if(model==null || model.isEmpty())
            return ""
        else
            return model.joinToString (separator = separator){ it }
    }

    override fun getModelValue(data: String?): List<String> {
        return data?.split(separator) ?: listOf()
    }
}
```

however following error is thrown during build phase:
`Error:error: *==========* :The specified custom TypeConverter's Model Value java.util.List<? extends java.lang.String> from com.kapuscinski.departures.persistence.db.StringListConverter must match the type of the column java.util.List<java.lang.String>.*==========*`

Am I missing something here, and how to fix this? Thanks in advance for help!

**Comment by agrosner** [comment]
Make the `List` mutable

**Comment by agrosner** [comment]
but also, you should avoid using `List` for columns completely. Rather you should just make it a `String` and manually pack and unpack the object like you do in the `TypeConverter` within the model itself.

**Comment by pokk** [comment]
DBFlow Version: 4.0.5

I have the same problem here.

This is my converter.
```kotlin
@com.raizlabs.android.dbflow.annotation.TypeConverter
class IntConverter: TypeConverter<String, MutableList<Int>>() {
    companion object {
        private const val SEPARATE_SYMBOL = ", "
    }

    override fun getModelValue(data: String): MutableList<Int> =
        data.split(SEPARATE_SYMBOL).map(String::toInt).toMutableList()

    override fun getDBValue(model: MutableList<Int>): String = model.joinToString()
}
```

This is my data class.
```kotlin
    @Table(database = MusicDatabase::class)
    data class MusicBean(
        @PrimaryKey(autoincrement = true) var id: Long = 0,
        @Column var uri: String = "",
        @Column var album_name: String = "",
        @Column var singer_name: String = "",
        @Column var song_name: String = "",
        @Column(typeConverter = IntConverter::class) var tag: MutableList<Int> = mutableListOf())
```

After I built, I got this msg from gradle console.

`e:     typeConverters.put(List<Integer>.class, new IntConverter());`

I checked the auto-generator method, I found `typeConverters.put(List<Integer>.class, new IntConverter());`

```java
public final class GeneratedDatabaseHolder extends DatabaseHolder {
  public GeneratedDatabaseHolder() {
    typeConverters.put(Boolean.class, new BooleanConverter());
    typeConverters.put(Character.class, new CharConverter());
    typeConverters.put(BigDecimal.class, new BigDecimalConverter());
    typeConverters.put(BigInteger.class, new BigIntegerConverter());
    typeConverters.put(Date.class, new SqlDateConverter());
    typeConverters.put(Time.class, new SqlDateConverter());
    typeConverters.put(Timestamp.class, new SqlDateConverter());
    typeConverters.put(Calendar.class, new CalendarConverter());
    typeConverters.put(GregorianCalendar.class, new CalendarConverter());
    typeConverters.put(java.util.Date.class, new DateConverter());
    typeConverters.put(List<Integer>.class, new IntConverter());
    typeConverters.put(UUID.class, new UUIDConverter());
    new MusicDatabaseMusicDatabase_Database(this);
  }
}
```

I couldn't make `MutableList` without generic type.

However I tried to use `IntArray` and `Array<Int>`, I could fix the problem.

How can we use `MutableList` in data class or do you have any feature to fix this problem for `MutableList`?

**Comment by mseroczynski** [comment]
Why is this issue closed, is it surely resolved? I'm able to reproduce this bug on 4.2.4
